src/cow-react/modules/limitOrders/containers/LimitOrdersWidget/index.tsx

```diff
@@ -26,35 +40,53 @@ export function LimitOrdersWidget() {
   }
   const inputCurrencyInfo: CurrencyInfo = {
     field: Field.INPUT,
-    currency: USDC[1],
-    rawAmount: null,
-    viewAmount: '1',
+    currency: inputCurrency,
+    rawAmount: inputCurrencyAmount,
+    viewAmount: inputCurrencyAmount?.toExact() || '',
     balance: null,
     fiatAmount: null,
     receiveAmountInfo: null,
   }
   const outputCurrencyInfo: CurrencyInfo = {
-    field: Field.INPUT,
-    currency: GNO[1],
-    rawAmount: null,
-    viewAmount: '2',
+    field: Field.OUTPUT,
+    currency: outputCurrency,
+    rawAmount: outputCurrencyAmount,
+    viewAmount: outputCurrencyAmount?.toExact() || '',
     balance: null,
     fiatAmount: null,
     receiveAmountInfo: null,
   }
-  const onCurrencySelection = () => {
-    //
-  }
-  const onUserInput = () => {
-    //
-  }
+  const onCurrencySelection = useCallback(
+    (field: Field, currency: Currency) => {
+      const inputCurrencyId = (field === Field.INPUT ? currency.symbol : inputCurrency?.symbol) || ''
+      const outputCurrencyId = (field === Field.OUTPUT ? currency.symbol : outputCurrency?.symbol) || ''
+      const route = parameterizeLimitOrdersRoute(chainId, inputCurrencyId, outputCurrencyId)
+
+      history.push(route)
+    },
+    [history, chainId, inputCurrency, outputCurrency]
+  )
+  const onUserInput = useCallback(
+    (field: Field, typedValue: string) => {
+      const inputCurrencyAmount = field === Field.INPUT ? typedValue : state.inputCurrencyAmount
+      const outputCurrencyAmount = field === Field.OUTPUT ? typedValue : state.outputCurrencyAmount
+
+      setState({ ...state, inputCurrencyAmount, outputCurrencyAmount })
```

Comment:
I would like to add. What if we make also the "smart" containers even dumber. 

We just abstract these things inside a custom hook. The custom hook can live in `/modules/limitOrders/containers/yourCustomHook`. Then is a matter of defining interfaces for these hooks as simple as possible. 

This will allow us to replace in the future the actual implementations on the hooks without having to even think about the react components that use it.

Reply:
fixed: https://github.com/cowprotocol/cowswap/pull/1110/commits/fbf16acae93ea0e1c4adec6437f5831433d8a043